include/snowflake/Simba_CRTFunctionSafe.h

```diff
@@ -21,186 +17,269 @@
 extern "C" {
 #endif
 
-#ifndef STDCALL
-    // match snowflake define in platform.h
-    #ifdef _WIN32
-        #define STDCALL __stdcall
-    #else
-        #define STDCALL
-    #endif
-#endif
-
-extern char* STDCALL sf_getenv(const char *name);
-extern char* STDCALL sf_strerror(int);
+#ifndef STDCALL
+    // match snowflake define in platform.h
+    #ifdef _WIN32
+        #define STDCALL __stdcall
+    #else
+        #define STDCALL
+    #endif
+#endif
+
+// Defined for #pragma warning messages.
+#define MACRO_TO_STRING2(x) #x
+#define MACRO_TO_STRING(x) MACRO_TO_STRING2(x)
 
-#if defined(WIN32) || defined(_WIN64)   // For Windows
+#if defined(_MSC_VER)
+#define SF_MACRO_DEPRECATED_WARNING(MSG) __pragma(message ( __FILE__ "(" MACRO_TO_STRING(__LINE__) ") : warning C4996: " MSG))
+#else
+#define SF_MACRO_DEPRECATED_WARNING(MSG)  _Pragma(MACRO_TO_STRING(GCC warning MSG))
+#endif
 
-    #define sf_free_s(x)        free(x)
```

Comment:
we are removing this define - should it be also deprecated?

Reply:
Will make it deprecated as well just in case.
This is added recently https://github.com/snowflakedb/libsnowflakeclient/pull/602 with the problematic implementation of sf_getenv and sf_strerror and should not be there in the first place.
I don't think external user could use it in proper way. It can only be used to free buffer returned from  sf_getenv/sf_strerror and can't be used as regular free (since it does nothing on non-Windows).

Reply:
done